docs/painless/painless-guide/painless-datetime.asciidoc

```diff
@@ -0,0 +1,295 @@
+----
++
+* Add three days to a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.plusDays(3);
+----
++
+* Subtract 125 minutes from a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.minusMinutes(125);
+----
++
+* Set the year on a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.withYear(1976);
+----
+
+==== Elapsed Time
+
+A script's objectives may require the elapsed time (difference) between two
+datetimes. Use either two longs or two ZonedDateTimes to calculate an elapsed
+time. Use <<subtraction-operator, subtraction>> to calculate an elapsed time
+between two longs. Use <<painless-api-reference-shared-ChronoUnit, ChronoUnit>>
+to calculate the difference between two ZonedDateTimes.
+
+===== Elapsed Time Examples
+
+* Elapsed time for two longs (milliseconds)
++
+[source,Painless]
+----
+long startTimestamp = 434931327000L;
+long endTimestamp = 434931330000L;
+long differenceInMillis = endTimestamp - startTimestamp;
+----
++
+* Elapsed time in milliseconds for two datetimes
++
+[source,Painless]
+----
+ZonedDateTime zdt1 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 11000000, ZoneId.of('Z'));
+ZonedDateTime zdt2 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 35, 0, ZoneId.of('Z'));
+long differenceInMillis = ChronoUnit.MILLIS.between(zdt1, zdt2);
+----
++
+* Elapsed time in days for two datetimes
++
+[source,Painless]
+----
+ZonedDateTime zdt1 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 11000000, ZoneId.of('Z'));
+ZonedDateTime zdt2 =
+        ZonedDateTime.of(1983, 10, 17, 22, 15, 35, 0, ZoneId.of('Z'));
+long differenceInDays = ChronoUnit.DAYS.between(zdt1, zdt2);
+----
+
+==== Datetime Comparison
+
+Comparisons of datetimes are often required to meet a script's objectives. Use
+either two longs or two ZonedDateTimes to do a datetime comparison. Use
+standard <<painless-operators-boolean, comparison operators>> to compare two
+long datetime representations. Use <<method-call-operator, method calls>> to
+compare two ZonedDateTime datetime representations.
+
+===== Datetime Comparison Examples
+
+* Comparison of two longs (milliseconds)
++
+[source,Painless]
+----
+long timestamp1 = 434931327000L;
+long timestamp2 = 434931330000L;
+
+if (timestamp1 > timestamp2) {
+   // handle condition
+}
+----
++
+* Before comparision of two datetimes
++
+[source,Painless]
+----
+ZonedDateTime zdt1 =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime zdt2 =
+        ZonedDateTime.of(1983, 10, 17, 22, 15, 35, 0, ZoneId.of('Z'));
+
+if (zdt1.isBefore(zdt2)) {
+    // handle condition
+}
+----
++
+* After comparision of two datetimes
```

Comment:
Maybe we could also have an example that compares a modification of one of the zdts? For example, finding if a time is at least 1 hour later than another

Reply:
My intention is to add more advanced examples in a follow up PR.  If it's all right I'd like to postpone this request until then.